ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -360,6 +361,27 @@ export class WorkflowDetails extends React.Component<RouteComponentProps<any>, W
         this.appContext.router.history.push(`${this.props.match.url}?${params.toString()}`);
     }
 
+    private renderCostOptimisations() {
+        const recommendations: string[] = [];
+        if (!this.state.workflow.spec.activeDeadlineSeconds) {
+            recommendations.push('activeDeadlineSeconds');
+        }
+        if (!this.state.workflow.spec.ttlStrategy) {
+            recommendations.push('ttlStrategy');
+        }
+        if (!this.state.workflow.spec.podGC) {
+            recommendations.push('podGC');
+        }
+        if (recommendations.length === 0) {
+            return;
+        }
+        return (
+            <CostOptimisationNudge name='workflow'>
+                You do not have {recommendations.join('/')} enabled for this workflow. Enabling these will reduce your costs.
```

Comment:
I think a link to the cost optimization docs here would be invaluable. I think we should wait until https://github.com/argoproj/argo/pull/2972 is merged and add the permalink to the doc here.

Reply:
<CostOptimisationNudge/> contains that - so this PR depends on that PR